### PR TITLE
Added emptymsgwidget test

### LINKED ISSUE #934: Tests: Create widget test for empty_list_message_widget.dart
Create widget test for https://github.com/dasfoo/delern/blob/master/flutter/lib/views/helpers/empty_list_message_widget.dart

Widget test doc: https://flutter.dev/docs/cookbook/testing/widget/introduction

### PR DESCRIPTION
**Pull request by jddeep**
Added emptymsgwidget test. Fixes #934

### CODE REVIEW

File: flutter/lib/views/helpers/empty_list_message_widget.dart [hunk 1]
```diff
@@ -8,9 +8,12 @@ class EmptyListMessageWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) => Center(
-        child: Text(
-          _displayText,
-          style: AppStyles.secondaryText,
+        child: Directionality(
+                  textDirection: TextDirection.ltr,
```
Comment (ksheremet) [inline flutter/lib/views/helpers/empty_list_message_widget.dart]: What about people who read from right to left, for example, from Arabic countries?

File: flutter/android/app/bin/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            <meta-data
+                android:name="io.flutter.app.android.SplashScreenUntilFirstFrame"
+                android:value="true" />
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN"/>
+                <category android:name="android.intent.category.LAUNCHER"/>
+            </intent-filter>
+        </activity>
+    </application>
+</manifest>
```
Comment (ksheremet) [inline flutter/android/app/bin/src/main/AndroidManifest.xml]: What IDE do you use? It looks like some files were just copied to ` flutter/android/app/bin/` folder. For Flutter it is better to use Android Studio, IntelliJ or VS Code.

This file is a copy of `delern/flutter/android/app/src/main/AndroidManifest.xml`

How about to do `rm -rf flutter/android/app/bin/` to get rid of duplicates?

File: flutter/android/.project [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
```
Comment (ksheremet) [inline flutter/android/.project]: It is Eclipse project files. No need to add them to PR.

File: flutter/android/app/.project [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
```
Comment (ksheremet) [inline flutter/android/app/.project]: It is Eclipse project files. No need to add them to PR.

File: flutter/android/app/bin/.project [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
```
Comment (ksheremet) [inline flutter/android/app/bin/.project]: It is Eclipse project files. No need to add them to PR.

File: flutter/android/app/bin/.settings/org.eclipse.buildship.core.prefs [hunk 6]
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=..
+eclipse.preferences.version=1
```
Comment (ksheremet) [inline flutter/android/app/bin/.settings/org.eclipse.buildship.core.prefs]: It is Eclipse project files. No need to add them to PR.

**Comment by jddeep** [comment]
@ksheremet please run it now and see. Thanks.

**Comment by jddeep** [comment]
@ksheremet now kindly check the PR. Thanks.

File: flutter/lib/views/helpers/empty_list_message_widget.dart [hunk 7]
```diff
@@ -8,9 +8,14 @@ class EmptyListMessageWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) => Center(
-        child: Text(
-          _displayText,
-          style: AppStyles.secondaryText,
+        child: Center(
+                  child: Directionality(
+                  textDirection: TextDirection.ltr,
```
Comment (ksheremet) [inline flutter/lib/views/helpers/empty_list_message_widget.dart]: Why is TextDirection.ltr used and not TextDirection.**rtl**?

There weren't any problems with the widget. The problem is with the test. It uses MateralApp components and doesn't wrap with MaterialApp widget

**Comment by jddeep** [comment]
@ksheremet i just looked into that. Kindly check it now.

File: flutter/lib/views/helpers/empty_list_message_widget.dart [hunk 8]
```diff
@@ -8,9 +8,9 @@ class EmptyListMessageWidget extends StatelessWidget {
...
   @override
   Widget build(BuildContext context) => Center(
-        child: Text(
-          _displayText,
-          style: AppStyles.secondaryText,
-        ),
+             child: Text(
+            _displayText,
+            style: AppStyles.secondaryText,
+          ),
```
Comment (ksheremet) [inline flutter/lib/views/helpers/empty_list_message_widget.dart]: It looks like the files have the wrong formatting. It is the last problem I see.

Could you please format files properly and rebase on gssoc19? Branch was updated.

For formatting please use IDE or `flutter format`. You can find more info about it [here](https://flutter.dev/docs/development/tools/formatting)

**Comment by jddeep** [comment]
@ksheremet so all changes have been made now.

**Comment by ksheremet** [comment]
@jddeep 
Almost there. 
The code in `flutter/test/empty_list_message_widget_test.dart` is not formated.

**Comment by jddeep** [comment]
@ksheremet Done. Thanks.

**Pull request merged**